File: backend/TravelPlanService/Models/Activity.cs
namespace TravelPlanService.Models;

public class Activity
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid TravelPlanId { get; set; }
    public Guid? DestinationId { get; set; }
    public string Name { get; set; } = string.Empty;
    public DateOnly Date { get; set; }
    public TimeOnly? Time { get; set; }
    public string? Location { get; set; }
    public string? Description { get; set; }
    public decimal? EstimatedCost { get; set; }
    public string Status { get; set; } = "Planned";
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public TravelPlan TravelPlan { get; set; } = null!;
    public Destination? Destination { get; set; }
}

File: frontend/src/models/Activity.js
export const ACTIVITY_STATUSES = ['Planned', 'Reserved', 'Completed', 'Cancelled'];

export class Activity {
  constructor(data = {}) {
    this.id = data.id || '';
    this.travelPlanId = data.travelPlanId || '';
    this.destinationId = data.destinationId || null;
    this.name = data.name || '';
    this.date = data.date || '';
    this.time = data.time || null;
    this.location = data.location || '';
    this.description = data.description || '';
    this.estimatedCost = data.estimatedCost || null;
    this.status = data.status || 'Planned';
    this.createdAt = data.createdAt || null;
  }
}
